Setting

fujisakiyuji opened the pull request:
#what
chat-space初めてのpull request

#why
作業前のセッティング

Commit:
Add to .gitignore

18行目：public / uploads / * 追記

Commit:
Add to README

ファイル内容の変更

hzak2xx commented:
readmeの記述がmembersテーブルに関するものだけになっているので、他も記述してWIPを外して提出してください

Commit:
Fix README.md file

各テーブル、カラム、アソシエーションの追加

fujisakiyuji commented:
追記致しました。

kohei-isoda commented on README.md:
```diff
@@ -1,24 +1,66 @@
+|title|string|null: false|
 
-* Database initialization
+### Association
```
throughオプションを指定してusersテーブルとアソシエーションを組みましょう！

fujisakiyuji replied:
captured_imagesのアソシエーションを削除し、throughを使用しmembersテーブルを追記しました。

kohei-isoda commented on README.md:
```diff
@@ -1,24 +1,66 @@
+- has_many :messages
+
+
+## captured_imagesテーブル
```
chatspaceではimage単体に何か処理を行う機能は実装いたしませんので、messagesテーブルにimage（captured_image）というカラムを作成していただければと思います！

fujisakiyuji replied:
captured_imagesテーブルを削除し、messagesテーブルにstring型でcaptured_imageカラムを追加致しました。

kohei-isoda commented on README.md:
```diff
@@ -1,24 +1,66 @@
+
+|Column|Type|Options|
+|------|----|-------|
+|member_id|integer|null: false, foreign_key: true|
```
usersテーブルでmember_idを持つことは難しいです。削除しましょう！
（理由が分からなければ周りの受講生に聞く or CONNECTで質問してみましょう！）

fujisakiyuji replied:
usersテーブルからmember_idを削除しました。

kohei-isoda commented on README.md:
```diff
@@ -1,24 +1,66 @@
+|Column|Type|Options|
+|------|----|-------|
+|member_id|integer|null: false, foreign_key: true|
+|group_id|integer|null: false, foreign_key: true|
```
こちらはmembersテーブルで管理しておりますので不要になります！

fujisakiyuji replied:
上記と内容と同じくgroup_idも削除しました。

kohei-isoda commented on README.md:
```diff
@@ -1,24 +1,66 @@
+|group_id|integer|null: false, foreign_key: true|
+|nickname|string|null: false|
+
+### Association
```
throughオプションを指定してusersテーブルとアソシエーションを組みましょう！

fujisakiyuji replied:
captured_imagesのアソシエーションを削除し、throughを使用しmembersテーブルを追記しました。

kohei-isoda reviewed (commented):
コメントいたしました！

Commit:
delete captured_images table

captured_imagesテーブルを削除しました。

Commit:
Add captured_image column to messages table

messagesテーブルにstring型でcaptured_imageカラムを追加致しました。

Commit:
Delete member_id column from users table

usersテーブルからmember_idカラムを削除しました。

Commit:
Delete group_id column from users table

usersテーブルからgroup_idカラムを削除しました。

Commit:
Modified the contents of the groups table

has_many :captured_imagesを削除
has_many :users, through: :membersを追記
上記の通り修正しました。

Commit:
Modified the contents of the users table

has_many :captured_imagesを削除
has_many :groups, through: :membersを追記
上記の通り修正しました。

fujisakiyuji commented:
ご指摘受けた箇所を全て修正しましたのでご確認の程よろしくお願いします。

kohei-isoda commented on README.md:
```diff
@@ -1,24 +1,51 @@
+
+|Column|Type|Options|
+|------|----|-------|
+|nickname|string|null: false|
```
chatspaceにはuserのnameを検索する機能がございますのでindexのオプションを貼りましょう！
また、uniqueでなければいけないので一意性制約を付与していただけると良いです！

fujisakiyuji replied:
ご指摘箇所にindexとuniqueを追記しました。
ご確認の程よろしくお願いします。

kohei-isoda reviewed (commented):
後もう少しです！

Commit:
Add to the nickname column of the users table

nicknameカラムにadd_index :users, nickname, :unique => trueを追記しました。

fujisakiyuji commented:
ご指摘箇所を修正しました。
ご確認お願いします。

miyagit commented on README.md:
```diff
@@ -1,24 +1,51 @@
-* Database creation
+|Column|Type|Options|
+|------|----|-------|
+|title|string|null: false|
```
nameで良いですよ

fujisakiyuji replied:
カラム名をnameに修正しました。
また、usersテーブルのカラム名もnicknameからnameに変更しました。
ご確認の程よろしくお願いします。

Commit:
Modify the groups table

groupsテーブルのカラム名titleをnameに変更しました。

Commit:
Modify the users table

usersテーブルのカラム名nicknameをnameに変更しました。
